compiler/src/Language/Bond/Codegen/Java/Class_java.hs

```diff
@@ -37,23 +228,157 @@ package #{javaPackage};
+        protected final void deserializeStructFields(com.microsoft.bond.BondType.TaggedDeserializationContext context, #{declName}#{params} value) throws java.io.IOException {
+            #{newlineSep 3 deserializeStructFieldDeclareBooleans structFields}
+            while (readField(context)) {
+                switch (context.readFieldResult.id) {
+                    #{newlineSep 5 deserializeStructFieldProcessFieldCases structFields}
+                }
+            }
+            #{newlineSep 3 deserializeStructFieldVerifyFields structFields}
+        }
+
+        @Override
+        public final void initializeStructFields(#{declName}#{params} value) {
+            #{newlineSep 3 initializeStructField structFields}
+        }
+    }
+
+    #{bondTypeStaticVariableDecl}
+
+    public static void initializeBondType() {
+        __StructBondTypeImpl.__StructBondTypeBuilderImpl.register();
+    }
+
+    static {
+        initializeBondType();
     }
+    #{bondTypeDescriptorInstanceVariableDecl}
 
-    #{doubleLineSep 1 publicField structFields}
+    #{doubleLineSep 1 publicFieldDecl structFields}
+
+    #{publicConstructorDecl}
+
+    @Override
+    public com.microsoft.bond.StructBondType<? extends com.microsoft.bond.BondSerializable> getBondType() {
+        return #{getBondTypeReturnValue};
+    }
 }|]
             where
                 interface = [lt| implements com.microsoft.bond.BondSerializable|]
                 params = angles $ sepBy ", " paramName declParams
+                declParamsWithIndex = zip [0 :: Int ..] declParams
                 baseClass x = [lt| extends #{javaType x}|]
-                javaDefault = Java.defaultValue java
-                publicField Field {..} = [lt|public #{javaType fieldType} #{fieldName} = #{javaDefault fieldType fieldDefault};|]
+                publicFieldDecl Field {..} = [lt|public #{javaType fieldType} #{fieldName};|]
+                fieldDescriptorFieldDecl Field {..} = [lt|private #{structFieldDescriptorTypeName java fieldType} #{fieldName};|]
+                fieldDescriptorFieldInit Field {..} = [lt|this.#{fieldName} = new #{structFieldDescriptorTypeName java fieldType}(#{initMethodParams});|]
+                        where
+                            initMethodParams = [lt|this#{fieldTypeParam}, #{fieldOrdinal}, "#{fieldName}", #{modifierConstantName fieldModifier}#{fieldDefaultValueParam}|]
+                                where
+                                    fieldTypeParam = if isGenericStructFieldDescriptor fieldType
+                                        then [lt|, #{structFieldDescriptorInitTypeExpr java fieldType}|]
+                                        else mempty
+                                    fieldDefaultValueParam = if not (isPrimitiveBondType fieldType) || (fieldDefault == Nothing)
+                                        then mempty
+                                        else [lt|, #{fieldDefaultValueInitParamExpr java fieldType fieldDefault}|]
```

Comment:
We're eight levels indented here - at the very least, field descriptors look complex enough to deserve their own top-level function.